common/build/eslint-plugin-fluid/src/test/mockFiles/no-unchecked-record-access/fileWithOnlyRecordAccess.ts

```diff
@@ -0,0 +1,78 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const record = { a: 1, b: 2 };
+const recordA = record.a; // ok: Accessing number property 'a' directly
+const recordB = record.b; // ok: Accessing number property 'b' directly
+
+interface NestedIndexProps {
+	nested: {
+		[key: string]: string;
+	};
+}
+type StaticType = { a: string; b: string };
+type IndexSignatureType = { [key: string]: string };
```

Comment:
Still don't see test cases where these are optional or values might be `undefined` or `null`.

Reply:
You should separate the additional cases. `| undefined` and `| null` are different than `| undefined | null`.